hoomd/md/integrate.py

```diff
@@ -51,6 +51,7 @@ def _attach_hook(self):
         if self.rigid is not None:
             self.rigid._attach(self._simulation)
         super()._attach_hook()
+        self.validate_group()
```

Comment:
```suggestion

    def _post_attach_hook(self):
        self.validate_groups()
```

Also, 1) add:
```
    def _post_attach_hook(self):
        """Hook called after applying parameter dicts."""
        pass
```
to _HOOMDBaseObject in `operation.py`, and 2) call `self._post_attach_hook()` at the end of `_HOOMDBaseObject.attach`

This set of changes will call `validate_groups` *after* the C++ `IntegratorTwoStep` instances is given the list of `methods`.